### DISCUSSION TITLE
SQLAlchemy does not support the health check capability for database connections in the connection pool.

### DISCUSSION THREAD

**Discussion by anueshenxing** [sqlalchemy/sqlalchemy#11566] [Usage Questions]
### Describe the bug

When using SQLAlchemy to access a database with multiple IP addresses (ip1, ip2, ip3), where ip1 is the primary node and ip2 and ip3 are the standby nodes, the current state of the database connection pool is: 10 idle connections and 5 active connections. 

If the primary node experiences a failure (e.g., network unreachable，or disk io block), and ip2 node takes over as the master, when attempting to obtain a new connection from the database connection pool, the initial attempts may still provide connections to the DB node at ip1, which are unusable. This process may need to be repeated until all idle connections have been attempted, after which a usable database connection from the new primary (ip2) can be obtained.

### SQLAlchemy Version in Use

1.4.29

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

PostgreSQL 12

### Python Version

3.9

### Operating system

centos

### To Reproduce

```python
NA
```

### Error

```
# Copy the complete stack trace and error message here, including SQL log output if applicable.
```

**Comment by CaselIT** [comment]
Hi,

I'm not sure what you expect sqlalchemy to do here. This behaviour seems like something that psycopg2 should provide.
